File: Project 3/Java/src/ga/Edge.java
package ga;

public class Edge implements Comparable<Edge>{

	private int start, end;
	private double weight;
	
	public Edge(int start, int end, double weight) {
		this.start = start;
		this.end = end;
		this.weight = weight;
	}
	
	public double getWeight() {
		return weight;
	}
	
	public int getStart() {
		return start;
	}
	
	public int getEnd() {
		return end;
	}

	@Override
	public int compareTo(Edge e) {
		double oWeight = e.getWeight();
		if(weight < oWeight) {
			return -1;
		} else if(weight > oWeight) {
			return 1;
		} else {
			return 0;
		}
	}
}

File: Project 3/Code/src/Edge.cpp
/*
 * Edge.cpp
 *
 *  Created on: Mar 7, 2017
 *      Author: Moan
 */

#include <math.h>
#include "Edge.h"
#include "Node.h"


Edge::Edge(Node &start, Node &end) {
	this->start = &start;
	this->end = &end;
	this->weight = calcWeight(start, end);
}

Node Edge::getStart() {
	return *start;
}
Node Edge::getEnd() {
	return *end;
}
double Edge::getWeight() {
	return weight;
}

double Edge::calcWeight(const Node &start, const Node &end) {
	int r1, r2, g1, g2, b1, b2;
	r1 = start.getR();
	r2 = end.getR();
	g1 = start.getG();
	g2 = end.getG();
	b1 = start.getB();
	b2 = end.getB();

	return sqrt(pow(r2-r1,2) + pow(g2-g1,2) + pow(b2-b1,2));
}

